scripts/build-pattern-library-index.js

```diff
@@ -0,0 +1,25 @@
+import { generateManifest } from '@hypothesis/frontend-build';
+import fs from 'fs';
+
+const templateMap = [
+  ['{{ css_bundle }}', 'styles/pattern-library.css'],
+  ['{{ js_bundle }}', 'scripts/pattern-library.bundle.js'],
+];
+
+export async function buildPatternLibraryIndex() {
+  const manifest = await generateManifest({
+    pattern: 'build/{scripts,styles}/pattern-library*.{css,js}',
+  });
+  const templateContent = fs
+    .readFileSync('templates/index.template.html')
+    .toString();
+
+  const processedTemplateContent = templateMap.reduce(
+    (content, [pattern, fileName]) => {
+      return content.replace(pattern, manifest[fileName] ?? pattern);
+    },
+    templateContent
+  );
+
+  fs.writeFileSync('build/index.html', processedTemplateContent);
```

Comment:
I considered introducing mustache back, but the use case is so simple that it felt unnecessary, but I'm open to do it. No strong opinion here.

Reply:
The two most important things IMO are:

1. It is obvious what is/is not a variable in the template
2. Appropriate escaping is applied

I think we can get away with fake templates for now, although it does mean that someone will have to know to change this if they decide to add other dynamic content in future.

Reply:
> although it does mean that someone will have to know to change this if they decide to add other dynamic content in future.

The main challenge here is mapping the manifest entries with their corresponding template variable, in a way that we don't couple the logic with the contents of the manifest itself (as it is above)

However, this is true even if we use a proper templating system. For example, with mustache this could look something like this:

```js
const templateMap = [
  ['css_bundle', 'styles/pattern-library.css'],
  ['js_bundle', 'scripts/pattern-library.bundle.js'],
];
const manifest = await generateManifest({
  pattern: 'build/{scripts,styles}/pattern-library*.{css,js}',
});
const templateContent = fs
  .readFileSync('templates/index.template.html')
  .toString();
const viewParams = templateMap.reduce(
    (view, [pattern, fileName]) => {
      view[pattern] = manifest[fileName] ?? pattern
      return view;
    },
    {}
  );

const processedTemplateContent = Mustache.render(templateContent, viewParams);

fs.writeFileSync('build/index.html', processedTemplateContent);
```

The obvious benefit of a templating system is that the replacement of variables is less fragile (extra/missing spaces and such).